hcmanager/utils.go

```diff
@@ -0,0 +1,136 @@
+package hcmanager
+
+import (
+	"context"
+	"errors"
+	"net"
+
+	hwcc "hardware-classification-controller/api/v1alpha1"
+
+	bmh "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	//LabelName initial name to the label key as hardware classification group
+	LabelName = "hardwareclassification.metal3.io/"
+
+	//Status extract the baremetal host for status ready
+	Status = "ready"
+
+	//DefaultLabel if label is missing from the Extracted Hardware Profile
+	DefaultLabel = "matches"
+
+	//CPULable label for extraction of hardware details
+	CPULable = "CPU"
+
+	//NICLable label for extraction of hardware details
+	NICLable = "NIC"
+
+	//DISKLable label for extraction of hardware details
+	DISKLable = "DISK"
+
+	//RAMLable label for extraction of hardware details
+	RAMLable = "RAM"
+)
+
+//FetchBmhHostList this function will fetch and return baremetal hosts in ready state
+func (mgr HardwareClassificationManager) FetchBmhHostList(Namespace string) ([]bmh.BareMetalHost, bmh.BareMetalHostList, error) {
+
+	ctx := context.Background()
+
+	bmhHostList := bmh.BareMetalHostList{}
+	var validHostList []bmh.BareMetalHost
+
+	opts := &client.ListOptions{
+		Namespace: Namespace,
+	}
+
+	// Get list of BareMetalHost from BMO
+	err := mgr.client.List(ctx, &bmhHostList, opts)
+	if err != nil {
+		return validHostList, bmhHostList, errors.New(err.Error())
+	}
+
+	// Get hosts in ready status from bmhHostList
+	for _, host := range bmhHostList.Items {
+		if host.Status.Provisioning.State == "ready" {
+			validHostList = append(validHostList, host)
+		}
```

Comment:
I am not clear why we would classify only the hosts in ready state ? I get why we should not try to classify a host that has not gone through introspection yet, but what about hosts that are already provisioned ? i.e. I change the HC CR while my BMHs are already deployed.

Reply:
Well. HCC relies on the host which has passed introspection successfully. That's the state we decided to go with first for this release. For provisioned hosts, not ready state hosts, we should come up with another proposal, because those types are not used for provisioning. 
If you have any case mind, we can certainly discuss on that and plan for support.